futures-util/src/lib.rs

```diff
@@ -54,6 +53,85 @@ macro_rules! delegate_sink {
     }
 }
 
+// FIXME: currently async/await is only available with std
+if_std! {
+    use std::marker::Unpin;
+    use std::mem::PinMut;
+
+    #[doc(hidden)]
+    pub fn assert_unpin<T: Future + Unpin>(_: &T) {}
+
+    /// A macro which returns the result of polling a future once within the
+    /// current `async` context.
+    ///
+    /// This macro is only usable inside of `async` functions, closures, and blocks.
+    #[macro_export]
+    macro_rules! poll {
```

Comment:
Maybe move these out of `lib.rs`

It would be more ideal if `lib.rs` had no definitions in it.

Reply:
What would you think about moving them all to `src/futures-util/await/*` to signify that they're macros that interact with async/await?

Reply:
Done.